core/mount/mount_linux.go

```diff
@@ -66,7 +66,12 @@ func prepareIDMappedOverlay(usernsFd int, options []string) ([]string, func(), e
 		return options, nil, fmt.Errorf("failed to parse overlay lowerdir's from given options")
 	}
 
-	tmpLowerdirs, idMapCleanUp, err := doPrepareIDMappedOverlay(lowerDirs, usernsFd)
+	tempRemountsLocation, err := os.MkdirTemp(tempMountLocation, "ovl-idmapped")
```

Comment:
Where are these directories deleted? The ownership of who has to do that is not clear. But I guess the cleanup function is deleting them?

Reply:
Yeah removed on line 271 in the cleanup callback. This didn't fundamentally change in this PR, I just moved the location where the directory is created from `doPrepareIDMappedOverlay` to `prepareIDMappedOverlay` to simplify testing `doPrepareIDMappedOverlay`